content/ADR-145-pick-realm-v2.md

```diff
@@ -59,46 +79,137 @@ https://github.com/decentraland/kernel/blob/main/packages/shared/dao/index.ts#L6
-          { maxAmountOfUsers: number;
-            maxCpu: number;
-            maxMemory: number;
-          }
+export function catalystVersionLink(configuration: CatalystVersionConfig) {
+  const minimumRequiredVersions = configuration
+
+  Object.keys(minimumRequiredVersions).forEach(([serviceName, serviceRequiredVersion]) => {
+    validCandidates = picked.filter(candidate => meetsVersion(candidate, serviceName, serviceRequiredVersion))
+    // ...
+  })
+
+  return validCandidates
 }
 ```
 
+_Configuration_:
 
-### Catalyst Version Link
+```typescript
+export type CatalystVersionConfig = {
+  config?: 
+          { comms?: string;
+            bff?: string;
+            content?: string;
+            lambdas?: string;
+          }
+}
+```
 
+### Force Catalyst
 
 _Description_:
 
-Each Catalyst (BFF) exposes the version of Content and Lambdas that it's using. The Kernel should read that field, compare them and prioritize the ones having the highest value.
-As Content Server and Lambdas share the same version for each Catalyst and they are using semver, there is always an unique way to sort the candidates.
-
+This rule will have multiple Catalysts, sorted by priority, defined in its configuration. The rule will attempt to connect to the first Catalyst in the list. If it is unavailable, it will continue trying with the subsequent Catalysts specified in the configuration in order of priority.
 
 _Pseudo Code_:
 
 ```typescript
-export function versionLink() {
-  const catalystsWithVersion: { realm: string, version: string } = peers.sortByVersion()
-  const versionBuckets: PriorityQueue<{ realm: string, version: string }[]> = groupByVersions(catalystsWithVersion)
-  // This selects randomly between the candidates in the selected bucket 
-  // And chooses the bucket with the same algorithm used for close peers
-  return selectFirstByScore(context, score, definitiveDecisionThreshold)
+export function catalystVersionLink(configuration: ForceCatalystConfig) {
+    let picked = context.picked
+    
+    const candidates = picked.map(candidate => candidate.catalystName)
+    const selected = configuration.sortedOptions?.find(candidate => candidates.includes(candidate))
+    
+    context.selected = !!selected 
+        ? picked.find(candidate => candidate.catalystName === selected)
+        : undefined
+
+    return context
 }
 ```
+
+_Configuration_:
+
+```typescript
+export type ForceCatalystConfig = {
+  config?: { sortedOptions?: [string]; }
+}
+```
+
+### Large Latency
+
+_Description_:
+
+This rule will still operate as it currently does. However, in order to implement it on the server-side, the client will need to check the latency on all nodes to pass this information to the service. To prevent clients from constantly calling all nodes to check latency, they will first check if the realm-picking service is using a `variant` that contains the `LARGE_LATENCY` filter and only if it does it will gather the latency information to pass it over.
```

Comment:
I'd change the responsibility here, instead of the client sending to the server their own latencies for each option, I'd prefer something like the server returning a recommended ordered list of catalysts and then the client will choose from that list according to its local latency.

For example:

```
Client ->  Server  // getListOfCatalyst() -> [catalystB, catalystD, catalystA]
Then the client chooses the first one on the list with an adequate latency
```